### ISSUE TITLE
radix lint unintentionally matches any function named parseInt

### ISSUE THREAD

**Issue by evmar** [palantir/tslint#2281]
### Bug Report

- __TSLint version__: 4.4.2
- __TypeScript version__: 2.1.6
- __Running TSLint via__: cli

#### TypeScript code being linted

```ts
namespace my {
  function parseInt(x: string) { ... }
}
my.parseInt('a');
```

with `tslint.json` configuration:

```json
    "radix": true,
```

#### Actual behavior
[Lint: tslint] Missing radix parameter

#### Expected behavior
Don't warn on parseInt unless it's the builtin one.

**Comment by alexsapps** [comment]
Sorry if obvious but just to be sure, it should work in this form too:
```
import {parseInt} from ...;
parseInt('a');
```

**Comment by adidahiya** [comment]
This is nontrivial because checking if it's the `parseInt` from lib.d.ts will require using the type checker. We could change the behavior to be act differently if this rule is run with type checking enabled, but for the most part we value fast syntactic checks that are accurate _most_ of the time over slow semantic checks that are accurate 100% of the time.

Related: https://github.com/palantir/tslint/issues/2091

**Comment by evmar** [comment]
I think syntactically you could just catch whether parseInt is a property access or not to cover 99% of cases.  It's pretty confusing to import a parseInt that shadows the global one.  I think this came up in some code where a parser class had `.parseString` and `.parseInt` and so on.

**Comment by ajafff** [comment]
#2352 fixes the problem with property access / method call

Shadowing `parseInt` with another function will still result in a failure (like the import example above). But that seems to be overkill for this rule. Other rules like `no-eval` also don't track shadowed indentifiers.
